docs/DeveloperGuide.md

```diff
@@ -349,106 +346,214 @@ The `ViewCompanyCommand` is implemented similarly.
-than attempting to perform the undo.
+[//]: # (### \[Proposed\] Undo/redo feature)
 
-</div>
+[//]: # ()
+[//]: # (#### Proposed Implementation)
 
-The following sequence diagram shows how the undo operation works:
+[//]: # ()
+[//]: # (The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo)
 
-![UndoSequenceDiagram](images/UndoSequenceDiagram.png)
+[//]: # (history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+[//]: # (following operations:)
 
-</div>
+[//]: # ()
+[//]: # (* `VersionedAddressBook#commit&#40;&#41;`— Saves the current address book state in its history.)
 
-The `redo` command does the opposite — it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once
-to the right, pointing to the previously undone state, and restores the address book to that state.
+[//]: # (* `VersionedAddressBook#undo&#40;&#41;`— Restores the previous address book state from its history.)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+[//]: # (* `VersionedAddressBook#redo&#40;&#41;`— Restores a previously undone address book state from its history.)
 
-</div>
+[//]: # ()
+[//]: # (These operations are exposed in the `Model` interface as `Model#commitAddressBook&#40;&#41;`, `Model#undoAddressBook&#40;&#41;`)
+
+[//]: # (and `Model#redoAddressBook&#40;&#41;` respectively.)
+
+[//]: # ()
+[//]: # (Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.)
+
+[//]: # ()
+[//]: # (Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the)
+
+[//]: # (initial address book state, and the `currentStatePointer` pointing to that single address book state.)
+
+[//]: # ()
+[//]: # (![UndoRedoState0]&#40;images/UndoRedoState0.png&#41;)
+
+[//]: # ()
+[//]: # (Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command)
+
+[//]: # (calls `Model#commitAddressBook&#40;&#41;`, causing the modified state of the address book after the `delete 5` command executes)
+
+[//]: # (to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book)
```

Comment:
If these are commented out, then I feel like we can delete it. What do you think?